src/shared/utils/exportData/exporters/ScheduleHistoryExporter.ts

```diff
@@ -514,7 +531,96 @@ export class ScheduleHistoryExporter extends DataExporter<
         );
       });
 
-    return applicableSchedules.sort((a, b) => {
+    const withResolvedScheduleAndAppletVersionLinking: ScheduleHistoryData[] = [];
+
+    /**
+     * When multiple schedules (default and individual) are linked to multiple applet versions on the same day,
+     * the default schedule should not show up for applet versions after the individual schedule is linked
+     */
+    Object.entries(groupBy(applicableSchedules, 'activityOrFlowId')).forEach(
+      ([_, entityGroupedSchedules]) => {
+        const typesOfSchedulesLinkedToday: Set<'default' | 'individual'> = new Set();
+        const appletVersionsLinkedToday: Set<string> = new Set();
+        const schedulesLinkedToday = entityGroupedSchedules.filter((schedule) => {
+          const scheduleLinkDate = DateTime.fromISO(schedule.linkedWithAppletAt, {
+            zone: 'UTC',
+          });
+
+          const linkedToday = scheduleLinkDate >= startOfTheDay && scheduleLinkDate <= endOfTheDay;
+
+          if (linkedToday) {
+            const type = !schedule.userId ? 'default' : 'individual';
+            typesOfSchedulesLinkedToday.add(type);
+            appletVersionsLinkedToday.add(schedule.appletVersion);
+          }
+
+          return linkedToday;
+        });
+
+        let resolvedSchedules = entityGroupedSchedules;
+        if (typesOfSchedulesLinkedToday.size > 1 && appletVersionsLinkedToday.size > 1) {
+          const sortedByLinkDate = [...schedulesLinkedToday].sort((a, b) => {
+            const aLinkDate = DateTime.fromISO(a.linkedWithAppletAt, { zone: 'UTC' });
+            const bLinkDate = DateTime.fromISO(b.linkedWithAppletAt, { zone: 'UTC' });
+
+            return aLinkDate.valueOf() - bLinkDate.valueOf();
+          });
+
+          const indexOfFirstIndividualScheduleLinkedToday = sortedByLinkDate.findIndex(
+            (schedule) => !!schedule.userId,
+          );
+          let indexOfLastIndividualScheduleLinkedToday = -1;
+          let indexOfLastDefaultScheduleLinkedTodayBeforeIndividual = -1;
+
+          for (let i = sortedByLinkDate.length - 1; i >= 0; i--) {
+            const schedule = sortedByLinkDate[i];
+            if (schedule.userId && indexOfLastIndividualScheduleLinkedToday === -1) {
+              indexOfLastIndividualScheduleLinkedToday = i;
+            } else if (!schedule.userId && indexOfLastIndividualScheduleLinkedToday !== -1) {
+              indexOfLastDefaultScheduleLinkedTodayBeforeIndividual = i;
+              break;
+            }
+          }
```

Comment:
This can be done a bit cleaner with a `reduce`: 
```suggestion
          const { lastIndividualIndex, lastDefaultBeforeIndividualIndex } = sortedByLinkDate.reduce(
            (acc, schedule, index) => {
              if (schedule.userId) {
                return { ...acc, lastIndividualIndex: index };
              } else if (
                !schedule.userId &&
                acc.lastIndividualIndex !== -1 &&
                acc.lastDefaultBeforeIndividualIndex === -1
              ) {
                return { ...acc, lastDefaultBeforeIndividualIndex: index };
              }

              return acc;
            },
            { lastIndividualIndex: -1, lastDefaultBeforeIndividualIndex: -1 },
          )
```

Reply:
Thanks for the suggestion @andrevitalb, but I don't think there's a reliable way to make this work with a forward reduce. The loop method is meant to run backwards so it's easy to find the last index of an individual event, and the index of the default event that came before it can only be set after its index has been found. With the reduce method, the index of the last individual schedule event will eventually settle on the correct one, but the default schedule event index probably won't. This is because it needs to know ahead of time what the index is for the individual schedule event, but that index keeps changing.

I think if I was to rewrite it as a reduce, I'd do it like this

```typescript
const { lastIndividualIndex, lastDefaultBeforeIndividualIndex } = [...sortedByLinkDate]
            .reverse()
            .reduce(
              (acc, schedule, index) => {
                if (schedule.userId && acc.lastIndividualIndex === -1) {
                  return { ...acc, lastIndividualIndex: sortedByLinkDate.length - 1 - index };
                } else if (
                  !schedule.userId &&
                  acc.lastIndividualIndex !== -1 &&
                  acc.lastDefaultBeforeIndividualIndex === -1
                ) {
                  return {
                    ...acc,
                    lastDefaultBeforeIndividualIndex: sortedByLinkDate.length - 1 - index,
                  };
                }

                return acc;
              },
              { lastIndividualIndex: -1, lastDefaultBeforeIndividualIndex: -1 },
            );
```

But I'm not sure we gain much readability from this, and we lose the ability to terminate the loop early when the two indexes have been found

Reply:
You're right. Seems like the way to go for now functionality-wise